src/QMCHamiltonians/NonLocalECPotential.cpp

```diff
@@ -306,6 +261,190 @@ void NonLocalECPotential::evaluate(ParticleSet& P, bool Tmove)
+      if (O.PPset[ipp])
+        O.PPset[ipp]->randomize_grid(*O.myRNG);
+
+    //loop over all the ions
+    const auto& myTable = P.getDistTable(myTableIndex);
+    // clear all the electron and ion neighbor lists
+    for (int iat = 0; iat < NumIons; iat++)
+      O.IonNeighborElecs.getNeighborList(iat).clear();
+    for (int jel = 0; jel < P.getTotalNum(); jel++)
+      O.ElecNeighborIons.getNeighborList(jel).clear();
+
+    if (ComputeForces)
+      APP_ABORT("NonLocalECPotential::mw_evaluateImpl(): Forces not imlpemented\n");
+    if (myTable.DTType != DT_SOA)
+      APP_ABORT("NonLocalECPotential::mw_evaluateImpl(): not imlpemented for AoS builds\n");
+
+    for (int ig = 0; ig < P.groups(); ++ig) //loop over species
+    {
+      auto& joblist = O.nlpp_jobs[ig];
+      joblist.clear();
+
+      for (int jel = P.first(ig); jel < P.last(ig); ++jel)
+      {
+        const auto& dist               = myTable.getDistRow(jel);
+        const auto& displ              = myTable.getDisplRow(jel);
+        std::vector<int>& NeighborIons = O.ElecNeighborIons.getNeighborList(jel);
+        for (int iat = 0; iat < O.NumIons; iat++)
+          if (O.PP[iat] != nullptr && dist[iat] < O.PP[iat]->getRmax())
+          {
+            NeighborIons.push_back(iat);
+            O.IonNeighborElecs.getNeighborList(iat).push_back(jel);
+            joblist.push_back(NLPPJob(iat, jel, P.R[jel], dist[iat], -displ[iat]));
+          }
+      }
+      // find the max number of jobs of all the walkers
+      max_num_jobs[ig] = std::max(max_num_jobs[ig], joblist.size());
+    }
+
+    O.Value = 0.0;
+  }
+
+  RefVector<NonLocalECPotential> ecp_potential_list;
+  RefVector<NonLocalECPComponent> ecp_component_list;
+  RefVector<ParticleSet> p_list;
+  std::vector<int> iat_list;
+  RefVector<TrialWaveFunction> psi_list;
+  std::vector<int> jel_list;
+  std::vector<RealType> r_list;
+  std::vector<PosType> dr_list;
+  std::vector<RealType> pairpots(nw);
+
+  ecp_potential_list.reserve(nw);
+  ecp_component_list.reserve(nw);
+  p_list.reserve(nw);
+  iat_list.reserve(nw);
+  psi_list.reserve(nw);
+  jel_list.reserve(nw);
+  r_list.reserve(nw);
+  dr_list.reserve(nw);
+
+  for (int ig = 0; ig < ngroups; ++ig) //loop over species
+  for (size_t jobid = 0; jobid < max_num_jobs[ig]; jobid++)
+  {
+    ecp_potential_list.clear();
+    ecp_component_list.clear();
+    p_list.clear();
+    iat_list.clear();
+    psi_list.clear();
+    jel_list.clear();
+    r_list.clear();
+    dr_list.clear();
+    for (size_t iw = 0; iw < nw; iw++)
+    {
+      NonLocalECPotential& O(static_cast<NonLocalECPotential&>(O_list[iw].get()));
+      ParticleSet& P(P_list[iw]);
+      if (jobid < O.nlpp_jobs[ig].size())
+      {
+        const auto& job = O.nlpp_jobs[ig][jobid];
+        ecp_potential_list.push_back(O);
+        ecp_component_list.push_back(*O.PP[std::get<0>(job)]);
+        p_list.push_back(P);
+        iat_list.push_back(std::get<0>(job));
```

Comment:
See comment in header.  If this isn't a Tuple but a struct this becomes less cryptic and error prone. Also looks like something in the debugger.